src/DotNetty.Transport/Channels/AbstractChannel.cs

```diff
@@ -738,7 +738,7 @@ protected virtual void Flush0()
                 this.inFlush0 = true;
 
                 // Mark all pending write requests as failure if the channel is inactive.
-                if (!this.channel.Active)
+                if (!this.CanWrite)
```

Comment:
that's just enough. If we see a bigger pattern in the future we'll rethink. For now it fits the bill.